packages/gatsby-source-drupal/src/gatsby-node.js

```diff
@@ -370,7 +373,44 @@ exports.sourceNodes = async (
           if (d.body.included) {
             dataArray.push(...d.body.included)
           }
-          if (d.body.links && d.body.links.next) {
+
+          // If JSON:API extras is configured to add the resource count, we can queue
+          // all API requests immediately.
+          if (d.body.meta?.count) {
+            // If we hadn't added urls yet
+            if (
+              d.body.meta.count > 50 &&
+              !typeRequestsQueued.has(d.body.data[0]?.type)
+            ) {
```

Comment:
What happens when the count < 50?

Reply:
This logic is run after the first collection API request has completed (we can't get the collection count until then). Since pages request up to 50 items, if the total count is < 50 then we don't need to fetch anymore.

Reply:
count could be undefined if not set in jsonapi_extras setting right?